lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1065,6 +1117,22 @@ def exec_module(self, **kwargs):
 
                         vm_resource.storage_profile.data_disks = data_disks
 
+                    # Before creating VM accept terms of plan if `accept_terms` is True
+                    if self.accept_terms:
+                        if not self.plan:
+                            self.fail("parameter error: plan must be specified and include name, product, and publisher")
```

Comment:
This isn't actually checking all those values though, would only pop the error if plan wasn't specified at all.